### PR TITLE
feat: shutting down when delete listener node

### LINKED ISSUE #4367: Implement handler for `tcp-listener delete` command
Currently, we have a stop listener command [here](https://github.com/build-trust/ockam/blob/2f445439fda29dd4a297f93b9f57f075b5edfbe3/implementations/rust/ockam/ockam_command/src/tcp/listener/delete.rs), but the corresponding [handler](https://github.com/build-trust/ockam/blob/2f445439fda29dd4a297f93b9f57f075b5edfbe3/implementations/rust/ockam/ockam_api/src/nodes/service/transport.rs#L91) doesn't support stopping a listener, since it wasn't possible before merging [this PR](https://github.com/build-trust/ockam/pull/4172). We would like to have that support. Ideally adding this functionality should come with corresponding bats test [here](https://github.com/build-trust/ockam/blob/2f445439fda29dd4a297f93b9f57f075b5edfbe3/implementations/rust/ockam/ockam_command/tests/bats/unit.bats)

---

_We love helping new contributors!_
_If you have questions or need help as you work on your first Ockam contribution, please leave a comment on [this discussion](https://github.com/ockam-network/ockam/discussions/1081)._
_If you're looking for other issues to contribute to, checkout [this discussion](https://github.com/build-trust/ockam/discussions/3206) and labels -  https://github.com/build-trust/ockam/labels/good%20first%20issue or https://github.com/build-trust/ockam/labels/help%20wanted_

### LINKED ISSUE #4464: create command for tcp-listener doesn't implement the body
### Observed behavior

related to this https://github.com/build-trust/ockam/issues/4419#issuecomment-1467679970

### Steps to reproduce

ockam tcp-listener create [addr]

### Expected behavior

post to Node Manager with body

### Ockam Version

latest

### PR DESCRIPTION
**Pull request by 0xkelvin**
## Checks

- [x] All commits in this Pull Request are [signed](https://docs.github.com/en/authentication/managing-commit-signature-verification/signing-commits) and Verified by Github.
- [x] All commits in this Pull Request follow the Ockam [commit message convention](https://github.com/build-trust/.github/blob/main/CONTRIBUTING.md#commit-messages).
- [x] I accept the Ockam Community [Code of Conduct](https://github.com/build-trust/.github/blob/main/CODE_OF_CONDUCT.md).
- [x] I have accepted the Ockam [Contributor License Agreement](https://github.com/build-trust/ockam-contributors/blob/main/CLA.md) by adding my Git/Github details in a row at the end of the [CONTRIBUTORS.csv](https://github.com/build-trust/ockam-contributors/blob/main/CONTRIBUTORS.csv) file in a separate pull request to the [build-trust/ockam-contributors](https://github.com/build-trust/ockam-contributors) repository. The easiest way to do this is to [edit the CONTRIBUTORS.csv](https://github.com/build-trust/ockam-contributors/edit/main/CONTRIBUTORS.csv) file in the github web UI and create a PR, this will mark the commit as verified.

### CODE REVIEW

**Comment by adrianbenavides** [comment]
Hey @0xkelvin, thanks for submitting another PR! Could you link the issue that this PR solves and add some description to it?

**Comment by 0xkelvin** [comment]
> Hey @0xkelvin, thanks for submitting another PR! Could you link the issue that this PR solves and add some description to it?

the PR is for https://github.com/build-trust/ockam/issues/4367
the expectation behaviour is the handler should shutdown the listener node when it got deleted 

cc : @adrianbenavides

File: implementations/rust/ockam/ockam_command/tests/bats/unit.bats
Comment (adrianbenavides) [inline implementations/rust/ockam/ockam_command/tests/bats/unit.bats:1]: You should add a new test, just as the `tcp connection - CRUD` but using the `listener` commands.

File: implementations/rust/ockam/ockam_api/src/nodes/service/transport.rs [hunk 1]
```diff
@@ -107,7 +107,8 @@ impl NodeManagerWorker {
 
         match node_manager.transports.get(&tid) {
             Some(t) if t.1 == TransportMode::Listen => {
-                warn!("It is not currently supported to destroy LISTEN transports");
+                // warn!("It is not currently supported to destroy LISTEN transports");
```
Comment (adrianbenavides) [inline implementations/rust/ockam/ockam_api/src/nodes/service/transport.rs]: You can remove this line if it's no longer needed.

**Review by adrianbenavides — CHANGES_REQUESTED** [review]
Looks great @0xkelvin ! Just a small comment about how to improve the test

**Review by adrianbenavides — APPROVED** [review]

**Comment by adrianbenavides** [comment]
For some reason, the https://github.com/build-trust/ockam/pull/4455 PR got merged before this one although this was added first to the merge queue. https://github.com/build-trust/ockam/pull/4455 includes the commits of the current PR, so I'm closing this one.

**Pull request closed without merging**